src/test/java/io/knotx/commons/http/request/AllowedHeadersFilterTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.commons.http.request;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class AllowedHeadersFilterTest {
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final String ACCEPT = "Accept";
+    private static final String[] HEADERS = {
+            ACCEPT,
+            "Accept-Encoding",
+            "Accept-Language",
+            "Connection",
+            "Host",
+            "Upgrade-Insecure-Requests",
+            "User-Agent",
+            "Keep-Alive"
+    };
+
+    private static List<Pattern> preparePatters() {
+        return Arrays.asList(HEADERS).stream().map(Pattern::compile).collect(Collectors.toList());
+    }
+
+    public static AllowedHeadersFilter getPatternFilter() {
+        return AllowedHeadersFilter.create(preparePatters());
```

Comment:
I am wondering about changing this interface from:
```
AllowedHeadersFilter create(List<Pattern> patterns)
```
to
```
AllowedHeadersFilter create(List<String> regexes)
```
Then we can remove all duplicated code responsible for creating patterns:
```
regexes.stream()
          .map(regex -> Pattern.compile(regex, Pattern.CASE_INSENSITIVE))
          .collect(Collectors.toList());
```
Then you can validate also:
- null
- empty list
- case insensitive

Reply:
@tomaszmichalak I like it.

Look at new commit.

I use:

```
AllowedHeadersFilter create(Set<String> regexes)
```

instead of:

```
AllowedHeadersFilter create(List<String> regexes)
```

Then we can use allowedRequestHeader instead of allowedRequestHeaderPatterns (after this changes we can delete it)

I add new:

```
AllowedHeadersFilter create(Set<String> regexes)
```

To make it more specific, I want to use it this way:

```
AllowedHeadersFilter.CaseInsensitive.create(strings);
```

While maintaining old implantation. Please give me feedback if I implemented this in wrong way.